tests/test_shadow_fork.py

```
@@ -72,19 +72,32 @@ def test_create_fails_if_already_active(self, tmp_path):
+    def test_create_falls_back_to_head_if_base_branch_missing(self, tmp_path):
+        """Test that create falls back to HEAD if base branch doesn't exist."""
         fork = ShadowFork(repo_root=tmp_path)
 
         with patch.object(fork, "_run_git") as mock_git:
-            mock_git.side_effect = [
-                (True, "main", ""),  # Get current branch
-                (True, "", ""),  # Check for uncommitted changes
-                (False, "", "fatal"),  # Base branch doesn't exist
-            ]
+            # We use a side_effect to handle variable number of calls
+            def side_effect(args, **kwargs):
```

Ruff (0.15.2)

[warning] Missing return type annotation for private function `side_effect`

(ANN202)